ledger/sandbox/src/main/scala/com/digitalasset/platform/apiserver/services/ApiSubmissionService.scala

```diff
@@ -211,25 +228,50 @@ final class ApiSubmissionService private (
           contractStore.lookupContractKey(commands.submitter, _),
           commands.commands
         )
-      submissionResult <- handleResult(res)
+      transactionInfo <- res.fold(error => {
+        Metrics.failedInterpretationsMeter.mark()
+        Future.failed(grpcError(toStatus(error)))
+      }, Future.successful)
+      partyAllocationResults <- if (configuration.implicitPartyAllocation) {
+        allocateParties(transactionInfo)
+      } else {
+        Future.successful(Seq.empty)
+      }
+      submissionResult <- partyAllocationResults.find(_ != SubmissionResult.Acknowledged) match {
+        case None => handleResult(transactionInfo)
+        case Some(result) => Future.successful(result)
+      }
     } yield submissionResult
 
-  private def handleResult(
-      res: scala.Either[ErrorCause, (SubmitterInfo, TransactionMeta, Transaction.Transaction)],
-  ) =
-    res match {
-      case Right((submitterInfo, transactionMeta, transaction)) =>
+  private def allocateParties(transactionInfo: TransactionInfo): Future[Seq[SubmissionResult]] = {
```

Comment:
I would suggest changing this to `allocateMissingInformees` or something similar, which is more specific.

Reply:
Done.